### PR TITLE
release notes for v1.5.8

### PR DESCRIPTION
**Pull request by jklynch**
This PR adds release notes for v1.5.8.

### CODE REVIEW

File: doc/source/api_changes.rst [hunk 1]
```diff
@@ -2,6 +2,14 @@
...
 =================
 
+v1.5.8 (2020-06-05)
+===================
+
+Fixed
+-----
+
+* Missing implementation details of :func:`~bluesky.run_engine.RunBundler.collect` have been added.
+
```
Comment (mrakitin) [inline doc/source/api_changes.rst:12]: Should we mention it's a backported change?

Comment (jklynch) [reply doc/source/api_changes.rst:12]: good idea

**Comment by jklynch** [comment]
test_failures_kill_run is failing unpredictably

**Comment by jklynch** [comment]
self-merged under cover of darkness

**Pull request merged**